lib/library.rb

```diff
@@ -0,0 +1,48 @@
+class Library
+  attr_reader :name, :books, :authors
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+  end
+
+  def add_author(author)
```

Comment:
This is a great use of the `!` on flatten!  You could also have a method for `books` that would iterate over authors and get all of their books on the fly, rather than holding this as an ivar.